hist/hist/inc/Math/WrappedMultiTF1.h

```diff
@@ -291,6 +302,65 @@ namespace ROOT {
          }
       }
 
+      // struct for dealing of generic Hessian computation, since it is available only in TFormula
+      template <class T>
+      struct GeneralHessianCalc {
+         static bool Hessian(TF1 *, const T *, const double *, T *)
+         {
+            Error("Hessian", "The vectorized implementation of ParameterHessian is not supported");
+            return false;
+         }
+         static bool IsAvailable(TF1 * ) { return false;}
+      };
+
+      template <>
+      struct GeneralHessianCalc<double> {
+         static bool Hessian(TF1 * func, const double *x, const double * par, double * h)
+         {
+            // compute Hessian if TF1 is a formula based
+            unsigned int np = func->GetNpar();
+            auto formula = func->GetFormula();
+            if (!formula) return false;
+            std::vector<double> h2(np*np);
+            func->SetParameters(par);
+            formula->HessianPar(x,h2);
+            for (unsigned int i = 0; i < np; i++) {
+               for (unsigned int j = 0; j <= i; j++) {
+                  unsigned int ih = j + i *(i+1)/2;  // formula for j <= i
+                  unsigned int im = i*np + j;
+                  h[ih] = h2[im];
+               }
+            }
+            return true;
+         }
+         static bool IsAvailable(TF1 * func) {
+            auto formula = func->GetFormula();
+            if (!formula) return false;
+            return formula->GenerateHessianPar();
+         }
+      };
+
+
+
+      template <class T>
+      bool WrappedMultiTF1Templ<T>::ParameterHessian(const T *x, const double *par, T *h) const
+      {
+         if (fLinear) {
+            unsigned int nh = NPar()*(NPar()+1)/2;
+            for (unsigned int i = 0; i < nh; i++) {
+               h[i] = 0;
+            }
```

Comment:
```suggestion
            std::fill(h, h + NPar()*(NPar()+1)/2, 0.0);
```
If you want a shorter alternative